pygmt/figure.py

```diff
@@ -401,89 +401,96 @@ def savefig(  # noqa: PLR0912
         if show:
             launch_external_viewer(str(fname))
 
-    def show(self, dpi=300, width=500, method=None, waiting=0.5, **kwargs):
+    def show(
+        self,
+        method: Literal["external", "notebook", "none", None] = None,
+        dpi: int = 300,
+        width: int = 500,
+        waiting: float = 0.5,
+        **kwargs,
+    ):
```

Comment:
Type hint output type as `None`?

```suggestion
    ) -> None:
```

Reply:
We didn't add the None type hints to other functions. If we decide to do it, then we can do it in a separate PR.